roles/docker_private_registry/tasks/main.yaml

```diff
@@ -0,0 +1,38 @@
+---
+#
+#  This role sets up and creates an insecure private registry on the localhost
+#  at port 5000. 
+#
+  - name: Fail if ansible_docker0.ipv4.address is not defined
+    fail: 
+      msg: "ansible_docker0.ipv4.address it not defined!"
+    when: ansible_docker0.ipv4.address is not defined
+    run_once: true
+
+  - name: edit /etc/sysconfig/docker file to add private registry
+    replace: 
+      dest=/etc/sysconfig/docker 
+      regexp='^ADD_REGISTRY=\'--add-registry registry.access.redhat.com\'' 
+      replace='ADD_REGISTRY=\'--add-registry {{ ansible_docker0.ipv4.address }}:5000 --add-registry registry.access.redhat.com\''
+
+  - name: edit /etc/sysconfig/docker file to enable insecure registry
+    replace:
+      dest=/etc/sysconfig/docker
+      regexp='^# INSECURE_REGISTRY=\'--insecure-registry\s?\''
+      replace='INSECURE_REGISTRY=\'--insecure-registry {{ ansible_docker0.ipv4.address }}:5000\''
+
+  - name: restart docker
+    service: name=docker state=restarted
+
+  - name: wait for docker restart to complete
+    pause: seconds=30
```

Comment:
You might be able to use `wait_for` to check to see if the Docker daemon is ready instead of sleeping.

http://docs.ansible.com/ansible/wait_for_module.html